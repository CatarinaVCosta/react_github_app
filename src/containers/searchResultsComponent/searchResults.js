import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import {Place, Email, Link as Linking} from '@material-ui/icons';
import { connect } from 'react-redux';
import SnackbarContent from '@material-ui/core/SnackbarContent';

function renderContent(value, icon) {
  return (
    value ?
      <div style={{ display: 'flex', flexDirection: 'row', justifyContent: 'center' }}>
        {icon}
        <Typography color="textSecondary">
          {value}
        </Typography>
      </div>
      : null
  )
}

class SearchResultsComponent extends React.Component {
  render() {
    return (
      <div>
        {this.props.user
        ? <Card>
          <CardContent>
            { this.props.user.avatar_url
              ? <div style={{ display: 'flex', justifyContent: 'center'}}>
                <img src={this.props.user.avatar_url} style={{ width: 200, height: 200 }} />
              </div>
              : null
            }
            {this.props.user.name
              ? <a href={this.props.user.html_url}>{this.props.user.name}</a>
              : null
            }
            {renderContent(this.props.user.location, <Place/>)}
            {renderContent(this.props.user.email, <Email/>)}
            {renderContent(this.props.user.html_url, <Linking/>)}
          </CardContent>
        </Card>
        : null }
        {this.props.errorMessage
          ? <div>
              <SnackbarContent
                style={{ color: 'red', justifyContent: 'center' }}
                message={this.props.errorMessage}
              />
            </div>
          : null
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.results.user,
    errorMessage: state.results.errorMessage
  }
};

export const SearchResults = connect(
  mapStateToProps
)(SearchResultsComponent);
