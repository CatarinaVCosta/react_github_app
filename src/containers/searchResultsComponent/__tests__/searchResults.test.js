import React from 'react';
import { SearchResults } from "../searchResults";
import configureStore from "redux-mock-store";
import { Provider } from 'react-redux';
import ReactTestRenderer from 'react-test-renderer';
import Typography from '@material-ui/core/Typography';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SnackbarContent from '@material-ui/core/SnackbarContent';

Enzyme.configure({ adapter: new Adapter() });

describe('Test search results', () => {
  const user = {
    avatar_url: 'dummy_avatar',
    name: 'dummy_name',
    location: 'dummy_location',
    email: 'dummy_email',
    html_url: 'dummy_url',
  };

  let store, errorStore;
  const mockStore = configureStore();

  let initialState = {
    results: {
      user: user,
      errorMessage: null
    }
  };

  let initialStateError = {
    results: {
      user: null,
      errorMessage: 'error'
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
    errorStore = mockStore(initialStateError)
  });


  it('Should match snapshot', () => {
    const wrapper = ReactTestRenderer.create(<Provider store={store}> <SearchResults /> </Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Img element src value should match with defined props', () => {
    const wrapper = shallow(<Provider store={store}> <SearchResults /> </Provider>);
    expect(wrapper.html()).toContain('src="dummy_avatar"')
  });

  it('href value should match with defined props', () => {
    const wrapper = shallow(<Provider store={store}> <SearchResults /> </Provider>);
    expect(wrapper.html()).toContain('href="dummy_url"')
  });

  it('Typography components values should match with defined props', () => {
    const wrapper = ReactTestRenderer.create(<Provider store={store}> <SearchResults /> </Provider>);
    const typographyLocation = wrapper.root.findAllByType(Typography)[0].props;
    const typographyEmail = wrapper.root.findAllByType(Typography)[1].props;
    const typographyUrl = wrapper.root.findAllByType(Typography)[2].props;

    expect(typographyLocation.children).toEqual(initialState.results.user.location);
    expect(typographyEmail.children).toEqual(initialState.results.user.email);
    expect(typographyUrl.children).toEqual(initialState.results.user.html_url);
  });

  it('Should not render Card component when props are undefined', () => {
    const wrapper = ReactTestRenderer.create(<Provider store={errorStore}> <SearchResults /> </Provider>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Should render SnackbarContent component when user not found', () => {
    const wrapper = ReactTestRenderer.create(<Provider store={errorStore}> <SearchResults /> </Provider>);
    const snackbar = wrapper.root.findAllByType(SnackbarContent)[0].props;

    expect(snackbar.message).toEqual(initialStateError.results.errorMessage);
  });
});
