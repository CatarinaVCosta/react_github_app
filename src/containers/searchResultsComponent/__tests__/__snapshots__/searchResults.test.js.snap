// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test search results Should match snapshot 1`] = `
Array [
  " ",
  <div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <img
            src="dummy_avatar"
            style={
              Object {
                "height": 200,
                "width": 200,
              }
            }
          />
        </div>
        <a
          href="dummy_url"
        >
          dummy_name
        </a>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            dummy_location
          </p>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            dummy_email
          </p>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            dummy_url
          </p>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;

exports[`Test search results Should not render Card component when props are undefined 1`] = `
Array [
  " ",
  <div>
    <div>
      <div
        className="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root MuiTypography-body2"
        role="alertdialog"
        style={
          Object {
            "color": "red",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="MuiSnackbarContent-message"
        >
          error
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
